src/plugins/gs-plugin-external-runtimes-cleaner.c

```diff
@@ -0,0 +1,239 @@
+/* -*- Mode: C; tab-width: 8; indent-tabs-mode: t; c-basic-offset: 8 -*-
+ *
+ * gs-plugin-external-runtimes-cleaner: This plugin handles the removal of no
+ * longer needed external apps' "external runtimes".
+ * It should be removed once the transition path to the Flatpak implementation
+ * of external apps is complete.
+ *
+ * Copyright (C) 2016 Endless Mobile, Inc
+ *
+ * Author: Joaquim Rocha <jrocha@endlessm.com>
+ *
+ * Licensed under the GNU General Public License Version 2
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include <config.h>
+#include <flatpak.h>
+#include <glib.h>
+#include <gnome-software.h>
+#include <string.h>
+
+#include "gs-appstream.h"
+#include "gs-flatpak.h"
+
+#define RUNTIME_INSTALLED_MTD_KEY "EndlessOS::legacy-ext-runtime-installed"
```

Comment:
Nit: use same define name you use in other plugin (or even better, move to a common header?)